searchguard/internalusers.py

```diff
@@ -29,17 +29,20 @@ def check_user_exists(username):
         raise CheckUserExistsException('Unknown error checking whether user {} exists'.format(username))
 
 
-def create_user(username):
+def create_user(username, password=None):
```

Comment:
Why not properties like modify_user? Because now you can define the password, but not the backendroles. I would suggest we use the properties argument and check whether the supplied JSON contains a password OR hash.

Reply:
I'm planning to add the backend roles in a different PR, I preferred the function signature to be easier to remember

```python
# use generated one
create_user(username)

# set explicit password
create_user(username, password)

# set properties
create_user(username, properties=properties)
# or with explicit password
create_user(username, password, properties)
```
Seems a decent function signature, as most user creation methods accept username/password as close arguments, then we'll inject the `password` in the properties passed on to REST API.

We can just accept the properties (like `modify_user`), then creating a user with an explicit password (but no intended options) would be like:

```
create_user(username, dict(password=password))
```